crates/gitbutler-branch-actions/src/dependencies.rs

```diff
@@ -73,6 +67,37 @@ pub fn compute_workspace_dependencies(
     })
 }
 
+/// Get the commits that need to be processed for the stack dependencies caculation.
+///
+/// Commit IDs are in the order that they are applied (parent first).
+/// Merge commits to the target branch are not included.
+fn get_commits_to_process(
+    repo: &git2::Repository,
+    stack: &Stack,
+    target_sha: &git2::Oid,
+) -> Result<Vec<git2::Oid>, anyhow::Error> {
+    let commit_ids = repo
+        .l(stack.head(), LogUntil::Commit(*target_sha), true)
+        .context("failed to list commits")?
+        .into_iter()
+        .rev()
+        .filter_map(|commit_id| {
+            let commit = repo.find_commit(commit_id).ok()?;
+            let has_integrated_parent = commit.parent_ids().any(|id| {
+                let (number_commits_ahead, _) = repo.graph_ahead_behind(id, *target_sha).unwrap();
+                number_commits_ahead == 0
+            });
+
+            if has_integrated_parent && commit.parent_count() > 1 {
+                None
+            } else {
+                Some(commit_id)
+            }
```

Comment:
We filter out the commits that have multiple parents (merge commits), out of which at least one is already integrated into the target branch.

Reply:
lets check if it's a merge commit before doing potentially expensive lookups